cpp/include/cugraph/graph_generators.hpp

```diff
@@ -76,10 +80,59 @@ std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>> generat
   uint64_t seed      = 0,
   bool clip_and_flip = false);
 
+/**
+ * @brief generate an edge list for an R-mat graph.
+ *
+ * This function allows multi-edges and self-loops similar to the Graph 500 reference
+ * implementation.
+ *
+ * NOTE: The scramble_vertex_ids function needs to be called in order to generate a
+ * graph conforming to the Graph 500 specification (note that scrambling does not
+ * affect cuGraph's graph construction performance, so this is generally unnecessary).
+ * If `edge_factor` is given (e.g. Graph 500), set @p num_edges to
+ * (size_t{1} << @p scale) * `edge_factor`. To generate an undirected graph, set @p b == @p c and @p
+ * clip_and_flip = true. All the resulting edges will be placed in the lower triangular part
+ * (including the diagonal) of the graph adjacency matrix.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param rng_state RAFT RNG state, updated with each call
```

Comment:
It looks like this is being passed a NULL ptr in tests. Is there a special behavior if this is NULL vs. a ptr to a RngState?

Reply:
My mistake, I did indeed comment in the wrong header file, and I see where the rng_state param is created/initialized before the call.